NitroxServer/Communication/Packets/Processors/StoryGoalExecutedProcessor.cs

```diff
@@ -0,0 +1,50 @@
+using NitroxModel.Packets;
+using NitroxServer.Communication.Packets.Processors.Abstract;
+using NitroxServer.GameLogic;
+using NitroxServer.GameLogic.Unlockables;
+
+namespace NitroxServer.Communication.Packets.Processors;
+
+public class StoryGoalExecutedProcessor : AuthenticatedPacketProcessor<StoryGoalExecuted>
+{
+    private readonly PlayerManager playerManager;
+    private readonly StoryGoalData storyGoalData;
+    private readonly ScheduleKeeper scheduleKeeper;
+    private readonly PDAStateData pdaStateData;
+
+    public StoryGoalExecutedProcessor(PlayerManager playerManager,  StoryGoalData storyGoalData, ScheduleKeeper scheduleKeeper, PDAStateData pdaStateData)
+    {
+        this.playerManager = playerManager;
+        this.storyGoalData = storyGoalData;
+        this.scheduleKeeper = scheduleKeeper;
+        this.pdaStateData = pdaStateData;
+    }
+
+    public override void Process(StoryGoalExecuted packet, Player player)
+    {
+        Log.Debug($"Processing StoryGoalExecuted: {packet}");
+        bool added = storyGoalData.CompletedGoals.Add(packet.Key);
+        switch (packet.Type)
+        {
+            case StoryGoalExecuted.EventType.RADIO:
+                if (added)
+                {
+                    storyGoalData.RadioQueue.Add(packet.Key);
```

Comment:
It is kinda weird that this class is maintaining internal sync/state across multiple `StoryGoalData` member variables.  Should this be moved to that class?

Reply:
This one part is a bit complex because the switch part is working with two members of ``StoryGoalData``, one from ``PDAStateData`` and even with ``ScheduleKeeper``. But we shouldn't break the present logic because it's issued from Subnautica's way of doing the same thing (``StoryGoal.Execute()``) and we need to replicate it correctly (also it's easier to understand how the sync/persistence is done if it's close to the base method).